### DISCUSSION TITLE
How do I use a different interface for the go.d/ping plugin?

### DISCUSSION THREAD

**Discussion by dosiek** [netdata/netdata#15056] [Q&A] [answered]
Hello, I just migrated over to a fresh install of Raspbian and installed netdata, previously I was using an old version with fping and I could change it by using a parameter "-I interfacename" in the fping_opts variable, however now after installing the go.d/ping plugin I don't see this type of option, any ideas on how to do it, and if it's even possible or do I have to dig around and install old netdata version with fping to get it running?

Thanks.

**Comment by ilyam8** [comment, marked as answer]
Hi, @dosiek. go.d.plugin uses https://github.com/prometheus-community/pro-bing ~~and I think it doesn't support setting the interface~~. Wrong, it [supports](https://github.com/prometheus-community/pro-bing/blob/a17ba032a911c7afdd16c0ef39278e603f2614b8/ping.go#L182-L183). We can add a new option to set it.

**Reply by ilyam8** [reply]
Added `interface` in https://github.com/netdata/go.d.plugin/pull/1179. See **Note** in the PR.

**Reply by dosiek** [reply]
oh cool! thanks. 
is it already up in the nightly build? what's the easiest way to update to it?

**Reply by ilyam8** [reply]
It is not. Will be in tomorrow's nightly build (you will get it on Wed).

**Reply by dosiek** [reply]
alright mate, thank you so much for the quick answer & help

**Reply by ilyam8** [reply]
I think there was a misunderstanding from my side, @dosiek.

> "-I interfacename

Is setting **outbound** interface. I was under the impression it is using a specific interface's IP address as the source of ICMP packets.

So, it is not possible to set an outbound interface right now. [prometheus-community/pro-bing](https://github.com/prometheus-community/pro-bing) doesn't support it, but there is a [PR](https://github.com/prometheus-community/pro-bing/pull/32) that adds this functionality.
